src/main/java/com/google/firebase/auth/FirebaseAuth.kt

```diff
@@ -259,10 +259,23 @@ class FirebaseAuth constructor(val app: FirebaseApp) : InternalAuthProvider {
         return source.task
     }
 
-    internal fun formatErrorMessage(title: String, request: Request, response: Response): String {
-        return "$title API returned an error, " +
-            "with url [${request.method()}] ${request.url()} ${request.body()} -- " +
-            "response [${response.code()}] ${response.message()} ${response.body().use { it?.string() }}"
+    internal fun createAuthInvalidUserException(
```

Comment:
I introduced this new method, because `response.body()` can be read only once, so I wanted to keep it in single method.